### PR TITLE
SYNR-1111 and SYNPY-629

### PR DESCRIPTION
**Pull request by brucehoff**

### CODE REVIEW

File: synapseclient/client.py [hunk 1]
```diff
@@ -611,7 +617,12 @@ def onweb(self, entity, subpageId=None):
 
 
     def printEntity(self, entity, ensure_ascii=True):
-        """Pretty prints an Entity."""
+        """
```
Comment (brucehoff) [inline synapseclient/client.py]: Ziming suggests we might deprecate printEntity

File: synapseclient/client.py [hunk 2]
```diff
@@ -2232,7 +2248,7 @@ def getProjectSetting(self, project, setting_type):
         Gets the ProjectSetting for a project
         :param project: Project entity or its id as a string
         :param setting_type: type of setting. Choose from: {'upload', 'external_sync', 'requester_pays'}
-        :return: The ProjectSetting as a dict or None if no settin of the specified type exist
+        :return: The ProjectSetting as a dict or None if no settings of the specified type exists.
```
Comment (brucehoff) [inline synapseclient/client.py]: should be exist

File: synapseclient/client.py [hunk 3]
```diff
@@ -2296,6 +2324,10 @@ def _findTeam(self, name):
     def getTeam(self, id):
         """
         Finds a team with a given ID or name.
+        
+        :param id:  The ID or name of the team to retrieve
+        
+        :return:  An object of type :py:class:`synapseclient.Team`
```
Comment (brucehoff) [inline synapseclient/client.py]: synapseclient.team.Team?  Make sure link works.

File: synapseclient/client.py [hunk 4]
```diff
@@ -2314,6 +2346,8 @@ def getTeam(self, id):
 
     def getTeamMembers(self, team):
         """
+        Lists the members of the given team.
+
```
Comment (brucehoff) [inline synapseclient/client.py]: :py:class:`synapseclient.team.TeamMember`? Make sure link works.

File: synapseclient/client.py [hunk 5]
```diff
@@ -2542,6 +2576,9 @@ def _getSubmissionBundles(self, evaluation, status=None, myOwn=False, limit=20,
 
     def getSubmissionBundles(self, evaluation, status=None, myOwn=False, limit=20, offset=0):
         """
+        Retrieve a paginated list of submission bundles (submission and submissions status) for an evaluation queue, optionally
```
Comment (brucehoff) [inline synapseclient/client.py]: Retrieve submission bundles...

File: synapseclient/client.py [hunk 6]
```diff
@@ -2600,7 +2637,10 @@ def _GET_paginated(self, uri, limit=20, offset=0):
 
     def getSubmission(self, id, **kwargs):
         """
-        Gets a :py:class:`synapseclient.evaluation.Submission` object.
+        Gets a :py:class:`synapseclient.evaluation.Submission` object by its id.
+        :param id:  The id of the submission to retrieve
```
Comment (brucehoff) [inline synapseclient/client.py]: need line in front of this

File: synapseclient/client.py [hunk 7]
```diff
@@ -2735,6 +2781,11 @@ def _storeWiki(self, wiki, createOrUpdate): # type: (Wiki, bool) -> Wiki
 
 
     def getWikiAttachments(self, wiki):
+        """
+        Retrieve the attachments to a wiki page.
+        :param wiki: the Wiki object for which the attachments are to be returned.
```
Comment (brucehoff) [inline synapseclient/client.py]: need line break above

File: synapseclient/client.py [hunk 8]
```diff
@@ -2785,6 +2836,10 @@ def getColumn(self, id):
         Gets a Column object from Synapse by ID.
 
         See: :py:mod:`synapseclient.table.Column`
+        
+        :param id: the ID of the column to retrieve
+        
+        :return: an object of type :py:class`synapseclient.table.Column`
```
Comment (brucehoff) [inline synapseclient/client.py]: missing colon

File: synapseclient/client.py [hunk 9]
```diff
@@ -2841,10 +2902,9 @@ def tableQuery(self, query, resultsAs="csv", **kwargs):
         """
         Query a Synapse Table.
 
-        :param query: query string in a `SQL-like syntax <http://docs.synapse.org/rest/org/sagebionetworks/repo/web/controller/TableExamples.html>`_::
-
+        :param query: query string in a `SQL-like syntax <http://docs.synapse.org/rest/org/sagebionetworks/repo/web/controller/TableExamples.html>`_:
```
Comment (brucehoff) [inline synapseclient/client.py]: space before trailing colon

File: synapseclient/client.py [hunk 10]
```diff
@@ -2841,10 +2902,9 @@ def tableQuery(self, query, resultsAs="csv", **kwargs):
...
             SELECT * from syn12345
-
+
```
Comment (brucehoff) [inline synapseclient/client.py]: move :return: to the bottom of the doc

File: synapseclient/client.py [hunk 11]
```diff
@@ -3062,9 +3122,9 @@ def createColumn(self, name, columnType, maximumSize=None, defaultValue=None, en
 
     def createColumns(self, columns):
         """
-        Creates a batch of py:class:`Column`s within a single request
-        :param columns: a list of py:class:`Column` objects
-        :return: a list of py:class:`Column` objects that have been created in Synapse
+        Creates a batch of :py:class:`Column`s within a single request
+        :param columns: a list of :py:class:`Column` objects
```
Comment (brucehoff) [inline synapseclient/client.py]: newline above

File: synapseclient/client.py [hunk 12]
```diff
@@ -2274,8 +2298,12 @@ def getEvaluationByName(self, name):
 
     def getEvaluationByContentSource(self, entity):
         """
-        Returns a generator over evaluations that
-        derive their content from the given entity
+        Returns a generator over evaluations that derive their content from the given entity
+        
+        :param entity:  The :py:class:`synapseclient.Project` whose Evaluations are to be fetched.
+        
+        :return: a Generator over the :py:class:`synapseclient.evaluation.Evaluation` objects for the given :py:class:`synapseclient.Project`
```
Comment (brucehoff) [inline synapseclient/client.py]: synapseclient.Project -> synapseclient.entity.Project ??

File: synapseclient/entity.py [hunk 13]
```diff
@@ -455,7 +455,12 @@ class Project(Entity):
     Projects in Synapse must be uniquely named. Trying to create a project with
     a name that's already taken, say 'My project', will result in an error
 
-    ::
+    :param name:            The name of the project
+    :param properties:      A map of Synapse properties
+    :param annotations:     A map of user defined annotations
+    :param local_state:     Allow local state to be given. This state information is not persisted  in the Synapse Repository.
```
Comment (brucehoff) [inline synapseclient/entity.py]: :param local_state: internal use only

File: synapseclient/entity.py [hunk 14]
```diff
@@ -474,8 +479,13 @@ class Folder(Entity):
     Represents a folder in Synapse.
 
     Folders must have a name and a parent and can optionally have annotations.
+    :param name:            The name of the folder
+    :param parent:          The parent project or folder
+    :param properties:      A map of Synapse properties
+    :param annotations:     A map of user defined annotations
+    :param local_state:     Allow local state to be given.  This state information is not persisted in the Synapse Repository.
```
Comment (brucehoff) [inline synapseclient/entity.py]: :param local_state: internal use only

File: synapseclient/entity.py [hunk 15]
```diff
@@ -474,8 +479,13 @@ class Folder(Entity):
     Represents a folder in Synapse.
 
     Folders must have a name and a parent and can optionally have annotations.
+    :param name:            The name of the folder
```
Comment (brucehoff) [inline synapseclient/entity.py]: newline above

File: synapseclient/entity.py [hunk 16]
```diff
@@ -474,8 +479,13 @@ class Folder(Entity):
...
+    :param parent:          The parent project or folder
+    :param properties:      A map of Synapse properties
+    :param annotations:     A map of user defined annotations
+    :param local_state:     Allow local state to be given.  This state information is not persisted in the Synapse Repository.
 
-    ::
+    Example::
```
Comment (brucehoff) [inline synapseclient/entity.py]: syn.store(folder)

File: synapseclient/entity.py [hunk 17]
```diff
@@ -492,12 +502,16 @@ class Link(Entity):
...
 
-    :param targetVersion:       Version of the file you want to create a link for
-
-    ::
+    :param targetId:        The ID of the entity to be linked
+    :param targetVersion:   The version of the entity to be linked
+    :param parent:          The parent project or folder
+    :param properties:      A map of Synapse properties
+    :param annotations:     A map of user defined annotations
+    :param local_state:     Allow local state to be given. This state information is not persisted  in the Synapse Repository.
```
Comment (brucehoff) [inline synapseclient/entity.py]: internal use only

File: synapseclient/entity.py [hunk 18]
```diff
@@ -529,12 +543,16 @@ class File(Entity, Versionable):
...
-                             Defaults to True (file should be uploaded)
+    :param synapseStore:     Whether the File should be uploaded or if only the path should be stored when :py:func:`synapseclient.Synapse.store` 
+                            is called on the File object. Defaults to True (file should be uploaded)
     :param contentType:      Manually specify Content-type header, for example "application/png" or "application/json; charset=UTF-8"
     :param dataFileHandleId: Defining an existing dataFileHandleId will use the existing dataFileHandleId
                              The creator of the file must also be the owner of the dataFileHandleId to have permission to store the file
-    ::
+    :param properties:      A map of Synapse properties
+    :param annotations:     A map of user defined annotations
+    :param local_state:     Allow local state to be given. This state information is not persisted  in the Synapse Repository.
```
Comment (brucehoff) [inline synapseclient/entity.py]: internal use only

File: synapseclient/entity.py [hunk 19]
```diff
@@ -631,7 +649,12 @@ class DockerRepository(Entity):
     To upload a docker image that is managed by Synapse please use the official Docker client and read http://docs.synapse.org/articles/docker.html for instructions on uploading a Docker Image to Synapse
     
     :param repositoryName: the name of the Docker Repository. Usually in the format: [host[:port]/]path. If host is not set, it will default to that of DockerHub. port can only be specified if the host is also specified
+    :param parent: the parent project for the Docker repository
+    :param properties:      A map of Synapse properties
+    :param annotations:     A map of user defined annotations
+    :param local_state:     Allow local state to be given.  This state information is not persisted in the Synapse Repository.
     
+    :return:  an object of type :py:class:`org.sagebionetworks.repo.model.docker.DockerRepository`
```
Comment (brucehoff) [inline synapseclient/entity.py]: synapseclient.entity.DockerRepository

File: synapseclient/table.py [hunk 20]
```diff
@@ -544,14 +544,18 @@ def _before_synapse_store(self, syn):
 
 class Schema(SchemaBase):
     """
-    A Schema is a :py:class:`synapse.entity.Entity` that defines a set of columns in a table.
+    A Schema is an :py:class:`synapse.entity.Entity` that defines a set of columns in a table.
```
Comment (brucehoff) [inline synapseclient/table.py]: synapseclient.entity.Entity

File: synapseclient/table.py [hunk 21]
```diff
@@ -544,14 +544,18 @@ def _before_synapse_store(self, syn):
...
 
-    :param name: give the Table Schema object a name
+    :param name: the name for the Table Schema object
```
Comment (brucehoff) [inline synapseclient/table.py]: document description parameter

File: synapseclient/table.py [hunk 22]
```diff
@@ -544,14 +544,18 @@ def _before_synapse_store(self, syn):
...
     :param description:
     :param columns: a list of :py:class:`Column` objects or their IDs
-    :param parent: the project (file a bug if you'd like folders supported) in Synapse to which this table belongs
-
-    ::
+    :param parent: the project in Synapse to which this table belongs
+    :param properties:      A map of Synapse properties
+    :param annotations:     A map of user defined annotations
+    :param local_state:     Allow local state to be given.
```
Comment (brucehoff) [inline synapseclient/table.py]: internal use only

File: synapseclient/table.py [hunk 23]
```diff
@@ -572,21 +576,24 @@ class EntityViewSchema(SchemaBase):
...
                                         the names in this list will not be automatically added as columns to the EntityViewSchema
                                         if they exist in any of the defined scopes.
-    The default columns will be added after a call to :py:meth:`synapseclient.Synapse.store`.
-    ::
-
-
-        project_or_folder = syn.get("syn123")
+    :param properties:      A map of Synapse properties
+    :param annotations:     A map of user defined annotations
+    :param local_state:     Allow local state to be given.  This state information is not persisted in the Synapse Repository.
```
Comment (brucehoff) [inline synapseclient/table.py]: internal use only

File: synapseclient/table.py [hunk 24]
```diff
@@ -572,21 +576,24 @@ class EntityViewSchema(SchemaBase):
...
+    :param columns: a list of :py:class:`Column` objects or their IDs. These are optional.
     :param parent: the project in Synapse to which this table belongs
     :param scopes: a list of Projects/Folders or their ids
-    :param view_type: the type of EntityView to display: either 'file' or 'project'. Defaults to 'file'
-    :param addDefaultViewColumns: If true adds all default columns (e.g. name, createdOn, modifiedBy etc.) Defaults to True.
-    :param addAnnotationColumns: If true adds columns for all annotation keys defined across all Entities in the EntityViewSchema's scope. Defaults to True.
+    :param type: the type of EntityView to display: either 'file','project' or 'file_and_table'. Defaults to 'file'.
+    :param addDefaultViewColumns: If true, adds all default columns (e.g. name, createdOn, modifiedBy etc.) Defaults to True.
+     The default columns will be added after a call to :py:meth:`synapseclient.Synapse.store`.
+    :param addAnnotationColumns: If true, adds columns for all annotation keys defined across all Entities in the EntityViewSchema's scope. Defaults to True.
```
Comment (brucehoff) [inline synapseclient/table.py]: annotation columns will be added after a call to :py:meth:`synapseclient.Synapse.store`.

File: synapseutils/monitor.py [hunk 25]
```diff
@@ -20,30 +20,29 @@ def notifyMe(syn, messageSubject='', retries=0):
     :param retries: Number of retries to attempt on failure (default=0)
 
     Example::
```
Comment (zimingd) [inline synapseutils/monitor.py]: The generated html is not showing a code block here. Need to add a newline after `Example::`

File: synapseclient/table.py [hunk 26]
```diff
@@ -572,21 +584,25 @@ class EntityViewSchema(SchemaBase):
     """
     A EntityViewSchema is a :py:class:`synapseclient.entity.Entity` that displays all files/projects (depending on user choice) within a given set of scopes
```
Comment (zimingd) [inline synapseclient/table.py]: For EntityViewSchema's documentation could you also rename `view_type` ->`type`? This documentation error fix is for [ SYNPY-629](https://sagebionetworks.jira.com/browse/SYNPY-629)

Comment (brucehoff) [reply synapseclient/table.py]: @zimingd that parameter had already been renamed in the documentation

**Pull request merged**